libs/react-helpers/src/hooks/use-mutation-observer.ts

```diff
@@ -0,0 +1,45 @@
+import { captureException } from '@sentry/react';
+import debounce from 'lodash/debounce';
+import { useEffect, useState } from 'react';
+
+type MutationObserverConfiguration = {
+  debounceTime: number;
+  config: MutationObserverInit;
+};
+
+const DEFAULT_OPTIONS: MutationObserverConfiguration = {
+  debounceTime: 1000 / 16,
```

Comment:
suggestion (non blocking): Presumably this is to get around 60fps? Perhaps use requestionAnimationFrame, though it might be more complex to make debounce time configurable

Reply:
Ah that's a leftover from the tests I've done it should be defaulted to 0 in order to cover 60fps because `MutationObserver` fires its changes in a microtask which gets being processed before the next repaint (same behaviour as in raf). Hence the debounce here is only for the operations that don't need to be processed as often.